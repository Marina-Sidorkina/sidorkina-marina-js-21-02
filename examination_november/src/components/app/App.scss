.app {
  box-sizing: border-box;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  max-width: 2000px;
  min-width: 320px;
  min-height: 100vh;
  padding-bottom: 60px;

  &_dark {
    background-color: #0d1117;
  }
}
