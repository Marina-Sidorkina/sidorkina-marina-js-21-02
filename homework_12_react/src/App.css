.App {
  position: relative;
  width: 1024px;
  min-height: 100vh;
  margin: 0 auto;
}

.main {
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 240px 1fr;
  grid-column-gap: 15px;
  width: 1024px;
  min-height: calc(100vh - 150px);
  padding: 15px;
}

.sidebar {
  box-sizing: border-box;
  width: 240px;
  padding-top: 35px;
  background-color: #f9ecf9;
  border-radius: 10px;
}

.filter {
  border: none;
}

.filter__category {
  margin-bottom: 10px;
  font-size: 20px;
  font-weight: bold;
}

.filter__values {
  list-style: none;
  padding: 0;
  margin: 0;
}

.filter__value {
  margin-bottom: 10px;
}

.filter__input {
  margin-right: 15px;
}

.filter__label {
  font-size: 18px;
}

.promo {
  margin-bottom: 20px;
  padding-left: 10px;
  padding-top: 20px;
}

.promo__title {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 32px;
}

.promo__message {
  margin-top: 5px;
  margin-bottom: 0;
  font-size: 18px;
}

.categories {
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 15px;
  margin-bottom: 35px;
}

.category {
  padding: 15px;
  background-color: #f9ecf9;
  border-radius: 10px;
}

.category__link {
  color: darkmagenta;
}

.category__title {
  margin-top: 0;
  margin-bottom: 10px;
  color: darkmagenta;
}

.category__title:hover {
  opacity: 0.7;
}

.category__message {
  margin-top: 0;
  margin-bottom: 0;
}

.favourite__title {
  margin-top: 0;
  margin-bottom: 25px;
  padding-left: 15px;
  font-size: 24px;
}

.pianos {
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 15px;
  grid-row-gap: 15px;
  padding: 0;
  margin: 0;
}

.piano {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-content: flex-start;
  justify-content: space-between;
  flex-wrap: wrap;
  height: 100px;
  padding: 15px;
  background-color: #f9ecf9;
  border-radius: 10px;
}

.piano__img {
  margin-right: 15px;
}

.piano__title-link {
  color: darkmagenta;
}

.piano__title {
  margin: 2px 0 5px 0;
  font-size: 16px;
  color: darkmagenta;
}

.piano__title:hover {
  opacity: 0.7;
}

.piano__button {
  box-sizing: border-box;
  width: 120px;
  height: 30px;
  color: white;
  line-height: 30px;
  text-decoration: none;
  text-align: center;
  background-color: darkmagenta;
  border-radius: 10px;
}

.piano__button:hover {
  opacity: 0.7;
}
