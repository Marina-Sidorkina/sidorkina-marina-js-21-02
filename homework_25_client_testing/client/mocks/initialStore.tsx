const initialStore = {
  userModal: {
    isOpened: false,
    values: {
      picture: '',
      name: 'John Doe',
      gender: 'male',
      dateOfBirth: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      registerDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      id: '12345',
      email: 'test',
      phone: '55555'
    }
  },
  login: {
    data: {
      error: false,
      isLoading: false,
      inputValue: '',
      authorizedUserId: '123345',
      authorizedUserName: 'Name',
      authorizedUserPicture: ''
    }
  },
  registration: {
    values: {
      name: '',
      gender: '',
      dateOfBirth: '',
      email: '',
      phone: '',
    },
    isLoading: false,
    error: false
  },
  languageSelector: {
    value: 'ru'
  },
  postModal: {
    isOpened: false,
    currentPostId: '12345'
  },
  userPosts: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  postsList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  usersList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      users: [{
        firstName: 'John',
        lastName: 'Doe',
        title: 'mr',
        id: '12345'
      }]
    }
  },
  userInfo: {
    data: {
      isLoading: false,
      user: {
        firstName: 'John',
        lastName: 'Doe',
        picture: '',
        title: 'mr',
        gender: 'male',
        dateOfBirth: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        registerDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        id: '12345',
        email: '',
        phone: '55555'
      }
    }
  },
  postModalPost: {
    isLoading: false,
    error: false,
    owner: {
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    },
    post: {
      id: '12345',
      image: '',
      text: 'text',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }
  },
  postModalComments: {
    page: 1,
    error: false,
    isLoading: false,
    comments: [{
      id: '1234',
      message: 'message',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }],
    owners: [{
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    }],
    limit: 6,
    totalComments: 0
  }
};

export default initialStore;
