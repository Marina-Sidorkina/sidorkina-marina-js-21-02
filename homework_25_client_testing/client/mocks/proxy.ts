import MockAdapter from 'axios-mock-adapter';
import { proxy } from '../src/api/proxy';

export const mock = new MockAdapter(proxy);
