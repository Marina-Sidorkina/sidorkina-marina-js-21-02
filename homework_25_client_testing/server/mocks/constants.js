module.exports = {
  errorText: 'Request failed with status code 520',
  successStatus: 200,
  errorStatus: 520
};
