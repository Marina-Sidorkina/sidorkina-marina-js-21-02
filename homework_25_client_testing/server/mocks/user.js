module.exports = {
  userListMockData: { data: [], page: 0, limit: 5, total: 100 },
  userMockData: {
    id: '60d0fe4f5311236168a109ce',
    title: 'mr',
    firstName: 'Rudi',
    lastName: 'Droste',
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg',
    gender: 'male',
    email: 'rudi.droste@example.com',
    dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787',
    location: {
      street: '1196, MÃ¼hlenweg',
      city: 'GrÃ¼nhain-Beierfeld',
      state: 'Sachsen-Anhalt',
      country: 'Germany',
      timezone: '+3:30'
    },
    registerDate: '2021-06-21T21:02:08.029Z',
    updatedDate: '2021-06-21T21:02:08.029Z'
  },
  userProcessedMockData: {
    id: '60d0fe4f5311236168a109ce',
    firstName: 'Rudi',
    lastName: 'Droste',
    title: 'mr',
    gender: 'male',
    email: 'rudi.droste@example.com',
    phone: '0421-7342787',
    dateOfBirth: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    registerDate: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg'
  },
  userPostsListMockData: {
    data: [{id: '12345', image: 'image', owner: {}, text: 'text'}],
    page: 0, limit: 1, total: 100
  },
  userUpdatedMockData: { firstName: 'Rudi', lastname: 'Droste', gender: 'male',
    email: 'rudi.droste@example.com', dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787'
  }
};
