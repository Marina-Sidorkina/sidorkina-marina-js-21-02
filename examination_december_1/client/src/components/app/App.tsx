import React, { useEffect } from 'react';
import i18next from 'i18next';
import {
  Route, Switch, HashRouter, Redirect
} from 'react-router-dom';
import styles from './App.module.scss';
import Header from '../headerComponents/header/Header';
import Footer from '../footerComponents/footer/Footer';
import Login from '../../pages/login/Login';
import Registration from '../../pages/registration/Registration';
import Posts from '../../pages/posts/Posts';
import Users from '../../pages/users/Users';
import Profile from '../../pages/profile/Profile';
import { ThemeContextProvider, ThemeContext } from '../../contexts/ThemeContext';
import { IThemeContextState } from '../../contexts/@types/themeContext';
import { RUSSIAN_LANGUAGE } from '../../constants/components';

function App() {
  useEffect(() => {
    i18next.changeLanguage(RUSSIAN_LANGUAGE);
  }, []);

  return (
    <ThemeContextProvider>
      <ThemeContext.Consumer>
        {(context: Partial<IThemeContextState>) => (
          <div className={!context.darkTheme ? styles.app : `${styles.app} ${styles.app_dark}`}>
            <HashRouter>
              <Header />
              <Switch>
                <Route path="/profile/:id">
                  <Profile />
                </Route>
                <Route path="/registration">
                  <Registration />
                </Route>
                <Route path="/login">
                  <Login />
                </Route>
                <Route path="/users">
                  <Users />
                </Route>
                <Route path="/posts">
                  <Posts />
                </Route>
                <Redirect from="/" to="/posts" />
              </Switch>
            </HashRouter>
            <Footer />
          </div>
        )}
      </ThemeContext.Consumer>
    </ThemeContextProvider>
  );
}

export default App;
