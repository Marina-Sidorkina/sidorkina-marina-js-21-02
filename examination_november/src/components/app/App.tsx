import React from 'react';
import './App.scss';
import Header from '../header/Header';
import Footer from '../footer/Footer';
import Posts from '../../pages/posts/Posts';

function App() {
  return (
    <div className="app">
      <Header />
      <Posts />
      <Footer />
    </div>
  );
}

export default App;
