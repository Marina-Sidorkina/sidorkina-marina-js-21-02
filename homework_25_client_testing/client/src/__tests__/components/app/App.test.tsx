import '../../../../mocks/matchMediaMock';
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import { Provider } from 'react-redux';
import App from '../../../components/app/App';
import mockStore from '../../../../mocks/mockStore';
import initialStore from '../../../../mocks/initialStore';

configure({ adapter: new Adapter() });

describe('App component test', () => {
  const store = mockStore(initialStore);
  store.dispatch = jest.fn();

  test('should render App', () => {
    const wrapper = mount(<Provider store={store}><App /></Provider>);
    expect(wrapper.length).toEqual(1);
  });

  test('should render App', () => {
    const wrapper = mount(<Provider store={store}><App /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
