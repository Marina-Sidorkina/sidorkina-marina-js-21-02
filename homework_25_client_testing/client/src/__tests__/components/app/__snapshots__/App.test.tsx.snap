// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test should render App 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App>
    <ThemeContextProvider>
      <div
        className="app"
      >
        <HashRouter>
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/posts",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <Header>
              <header
                className="header"
              >
                <div
                  className="container"
                >
                  <Logo>
                    <div
                      className="logo"
                    >
                      <img
                        alt="Логотип"
                        className="img"
                        src="logo.png"
                      />
                      <p
                        className="text"
                      >
                        Delta World
                      </p>
                    </div>
                  </Logo>
                  <Navigation>
                    <nav
                      className="navigation"
                    >
                      <ul
                        className="list"
                      >
                        <li
                          className="item"
                        >
                          <Link
                            className="link"
                            to="/users"
                          >
                            <LinkAnchor
                              className="link"
                              href="#/users"
                              navigate={[Function]}
                            >
                              <a
                                className="link"
                                href="#/users"
                                onClick={[Function]}
                              >
                                <NavigationUsersIcon
                                  dark={false}
                                >
                                  <svg
                                    className="svg svg_users"
                                    fill="none"
                                    strokeLinecap="round"
                                    strokeLinejoin="round"
                                    strokeWidth="2"
                                    viewBox="0 0 24 24"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M17 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
                                    />
                                    <circle
                                      cx="9"
                                      cy="7"
                                      r="4"
                                    />
                                    <path
                                      d="M23 21v-2a4 4 0 0 0-3-3.87"
                                    />
                                    <path
                                      d="M16 3.13a4 4 0 0 1 0 7.75"
                                    />
                                  </svg>
                                </NavigationUsersIcon>
                                <span
                                  className="text"
                                >
                                  Пользователи
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </li>
                        <li
                          className="item"
                        >
                          <Link
                            className="link"
                            to="/posts"
                          >
                            <LinkAnchor
                              className="link"
                              href="#/posts"
                              navigate={[Function]}
                            >
                              <a
                                className="link"
                                href="#/posts"
                                onClick={[Function]}
                              >
                                <NavigationPostsIcon
                                  dark={false}
                                >
                                  <svg
                                    className="svg svg_posts"
                                    height="30"
                                    viewBox="0 0 96 96"
                                    width="30"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M84 24v48c0 3.312-2.688 6-6 6h-63c0 3.312 2.688 6 6 6h63c3.312 0 6-2.688 6-6v-48c0-3.312-2.688-6-6-6zM81 69v-48c0-3.312-2.688-6-6-6h-63c-3.312 0-6 2.688-6 6v48c0 3.312 2.688 6 6 6h63c3.312 0 6-2.688 6-6zM12 21h63v48h-63v-48zM46.881 51.933l-8.964-9.543-8.967 15.312-6.726-7.029-7.287 15.387h57.165l-12.891-33.735-12.33 19.608zM24 42c3.312 0 6-2.688 6-6s-2.688-6-6-6-6 2.688-6 6 2.688 6 6 6z"
                                    />
                                  </svg>
                                </NavigationPostsIcon>
                                <span
                                  className="text"
                                >
                                  Посты
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </li>
                      </ul>
                    </nav>
                  </Navigation>
                  <Authorization>
                    <div
                      className="authorization"
                    >
                      <div
                        className="dropdown"
                      >
                        <input
                          className="toggle"
                          id="authorization__toggle"
                          type="checkbox"
                        />
                        <label
                          className="btn"
                          htmlFor="authorization__toggle"
                        >
                          <span
                            className="control"
                          />
                        </label>
                        <div
                          className="menu"
                        >
                          <AuthorizedUserBlock>
                            <div
                              className="userBlock"
                            >
                              <div
                                className="item"
                              >
                                <img
                                  alt="Фото пользователя"
                                  className="userPhoto"
                                  src=""
                                />
                                <Link
                                  className="link"
                                  to="/profile/123345"
                                >
                                  <LinkAnchor
                                    className="link"
                                    href="#/profile/123345"
                                    navigate={[Function]}
                                  >
                                    <a
                                      className="link"
                                      href="#/profile/123345"
                                      onClick={[Function]}
                                    >
                                      <span
                                        className="userName"
                                      >
                                        Name
                                      </span>
                                    </a>
                                  </LinkAnchor>
                                </Link>
                              </div>
                              <div
                                className="item"
                              >
                                <span
                                  className="logout"
                                  onClick={[Function]}
                                >
                                  выход
                                </span>
                              </div>
                            </div>
                          </AuthorizedUserBlock>
                        </div>
                      </div>
                    </div>
                  </Authorization>
                </div>
              </header>
            </Header>
            <Switch>
              <Route
                computedMatch={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/posts",
                    "url": "/posts",
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/posts",
                    "search": "",
                    "state": undefined,
                  }
                }
                path="/posts"
              >
                <Connect(Posts)>
                  <Posts
                    modalIsOpened={false}
                    page={0}
                    perPage={6}
                    total={100}
                    updatePage={[Function]}
                  >
                    <Main>
                      <main
                        className="main"
                      >
                        <div
                          className="container"
                        >
                          <PostsList>
                            <ul
                              className="list"
                            >
                              <li
                                key="12345"
                              >
                                <PostsListItem
                                  avatar=""
                                  date="05 October 2011 14:48 UTC"
                                  firstName="John"
                                  id="12345"
                                  image=""
                                  lastName="Doe"
                                  text="text"
                                  title="mr"
                                  userId="12345"
                                >
                                  <div
                                    className="item"
                                    onClick={[Function]}
                                  >
                                    <div
                                      className="userBlock"
                                    >
                                      <img
                                        alt="Аватар пользователя"
                                        className="userImg"
                                        src=""
                                      />
                                      <div
                                        className="user"
                                      >
                                        <ComponentWithHelper
                                          firstName="John"
                                          lastName="Doe"
                                          title="mr"
                                        >
                                          <div
                                            className="helper"
                                            onMouseOut={[Function]}
                                            onMouseOver={[Function]}
                                          >
                                            <PostsListItemName
                                              firstName="John"
                                              lastName="Doe"
                                              title="mr"
                                            >
                                              <div
                                                className="userName"
                                              >
                                                г-н John Doe
                                              </div>
                                            </PostsListItemName>
                                          </div>
                                        </ComponentWithHelper>
                                        <div
                                          className="date"
                                        >
                                          05 October 2011 14:48 UTC
                                        </div>
                                      </div>
                                    </div>
                                    <img
                                      alt="Пост пользователя"
                                      className="post"
                                      src=""
                                    />
                                    <p
                                      className="text"
                                    >
                                      text
                                    </p>
                                  </div>
                                </PostsListItem>
                              </li>
                            </ul>
                          </PostsList>
                          <Paginator
                            current={0}
                            modal={false}
                            onPageChange={[Function]}
                            perPage={6}
                            total={100}
                          >
                            <Pagination
                              className="paginator
        "
                              current={0}
                              onChange={[Function]}
                              pageSize={6}
                              showQuickJumper={true}
                              simple={true}
                              total={100}
                            >
                              <LocaleReceiver
                                componentName="Pagination"
                                defaultLocale={
                                  Object {
                                    "items_per_page": "/ page",
                                    "jump_to": "Go to",
                                    "jump_to_confirm": "confirm",
                                    "next_3": "Next 3 Pages",
                                    "next_5": "Next 5 Pages",
                                    "next_page": "Next Page",
                                    "page": "Page",
                                    "page_size": "Page Size",
                                    "prev_3": "Previous 3 Pages",
                                    "prev_5": "Previous 5 Pages",
                                    "prev_page": "Previous Page",
                                  }
                                }
                              >
                                <Pagination
                                  className="paginator
        "
                                  current={0}
                                  defaultCurrent={1}
                                  defaultPageSize={10}
                                  hideOnSinglePage={false}
                                  itemRender={[Function]}
                                  jumpNextIcon={
                                    <a
                                      className="ant-pagination-item-link"
                                    >
                                      <div
                                        className="ant-pagination-item-container"
                                      >
                                        <ForwardRef(DoubleRightOutlined)
                                          className="ant-pagination-item-link-icon"
                                        />
                                        <span
                                          className="ant-pagination-item-ellipsis"
                                        >
                                          •••
                                        </span>
                                      </div>
                                    </a>
                                  }
                                  jumpPrevIcon={
                                    <a
                                      className="ant-pagination-item-link"
                                    >
                                      <div
                                        className="ant-pagination-item-container"
                                      >
                                        <ForwardRef(DoubleLeftOutlined)
                                          className="ant-pagination-item-link-icon"
                                        />
                                        <span
                                          className="ant-pagination-item-ellipsis"
                                        >
                                          •••
                                        </span>
                                      </div>
                                    </a>
                                  }
                                  locale={
                                    Object {
                                      "items_per_page": "/ page",
                                      "jump_to": "Go to",
                                      "jump_to_confirm": "confirm",
                                      "next_3": "Next 3 Pages",
                                      "next_5": "Next 5 Pages",
                                      "next_page": "Next Page",
                                      "page": "Page",
                                      "page_size": "Page Size",
                                      "prev_3": "Previous 3 Pages",
                                      "prev_5": "Previous 5 Pages",
                                      "prev_page": "Previous Page",
                                    }
                                  }
                                  nextIcon={
                                    <button
                                      className="ant-pagination-item-link"
                                      tabIndex={-1}
                                      type="button"
                                    >
                                      <ForwardRef(RightOutlined) />
                                    </button>
                                  }
                                  onChange={[Function]}
                                  onShowSizeChange={[Function]}
                                  pageSize={6}
                                  prefixCls="ant-pagination"
                                  prevIcon={
                                    <button
                                      className="ant-pagination-item-link"
                                      tabIndex={-1}
                                      type="button"
                                    >
                                      <ForwardRef(LeftOutlined) />
                                    </button>
                                  }
                                  selectComponentClass={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "OptGroup": [Function],
                                      "Option": [Function],
                                      "SECRET_COMBOBOX_MODE_DO_NOT_USE": "SECRET_COMBOBOX_MODE_DO_NOT_USE",
                                      "render": [Function],
                                    }
                                  }
                                  selectPrefixCls="ant-select"
                                  showLessItems={false}
                                  showPrevNextJumpers={true}
                                  showQuickJumper={true}
                                  showTitle={true}
                                  simple={true}
                                  style={Object {}}
                                  total={100}
                                  totalBoundaryShowSizeChanger={50}
                                >
                                  <ul
                                    className="ant-pagination ant-pagination-simple paginator
        "
                                    style={Object {}}
                                  >
                                    <li
                                      aria-disabled={true}
                                      className="ant-pagination-prev ant-pagination-disabled"
                                      onClick={[Function]}
                                      onKeyPress={[Function]}
                                      tabIndex={null}
                                      title="Previous Page"
                                    >
                                      <button
                                        className="ant-pagination-item-link"
                                        disabled={true}
                                        tabIndex={-1}
                                        type="button"
                                      >
                                        <ForwardRef(LeftOutlined)>
                                          <AntdIcon
                                            icon={
                                              Object {
                                                "icon": Object {
                                                  "attrs": Object {
                                                    "focusable": "false",
                                                    "viewBox": "64 64 896 896",
                                                  },
                                                  "children": Array [
                                                    Object {
                                                      "attrs": Object {
                                                        "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                                      },
                                                      "tag": "path",
                                                    },
                                                  ],
                                                  "tag": "svg",
                                                },
                                                "name": "left",
                                                "theme": "outlined",
                                              }
                                            }
                                          >
                                            <span
                                              aria-label="left"
                                              className="anticon anticon-left"
                                              role="img"
                                            >
                                              <IconReact
                                                icon={
                                                  Object {
                                                    "icon": Object {
                                                      "attrs": Object {
                                                        "focusable": "false",
                                                        "viewBox": "64 64 896 896",
                                                      },
                                                      "children": Array [
                                                        Object {
                                                          "attrs": Object {
                                                            "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                                          },
                                                          "tag": "path",
                                                        },
                                                      ],
                                                      "tag": "svg",
                                                    },
                                                    "name": "left",
                                                    "theme": "outlined",
                                                  }
                                                }
                                              >
                                                <svg
                                                  aria-hidden="true"
                                                  data-icon="left"
                                                  fill="currentColor"
                                                  focusable="false"
                                                  height="1em"
                                                  key="svg-left"
                                                  viewBox="64 64 896 896"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                                                    key="svg-left-svg-0"
                                                  />
                                                </svg>
                                              </IconReact>
                                            </span>
                                          </AntdIcon>
                                        </ForwardRef(LeftOutlined)>
                                      </button>
                                    </li>
                                    <li
                                      className="ant-pagination-simple-pager"
                                      title="0/17"
                                    >
                                      <input
                                        onBlur={[Function]}
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        onKeyUp={[Function]}
                                        size="3"
                                        type="text"
                                        value={0}
                                      />
                                      <span
                                        className="ant-pagination-slash"
                                      >
                                        /
                                      </span>
                                      17
                                    </li>
                                    <li
                                      aria-disabled={false}
                                      className="ant-pagination-next"
                                      onClick={[Function]}
                                      onKeyPress={[Function]}
                                      tabIndex={null}
                                      title="Next Page"
                                    >
                                      <button
                                        className="ant-pagination-item-link"
                                        disabled={false}
                                        tabIndex={-1}
                                        type="button"
                                      >
                                        <ForwardRef(RightOutlined)>
                                          <AntdIcon
                                            icon={
                                              Object {
                                                "icon": Object {
                                                  "attrs": Object {
                                                    "focusable": "false",
                                                    "viewBox": "64 64 896 896",
                                                  },
                                                  "children": Array [
                                                    Object {
                                                      "attrs": Object {
                                                        "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                                      },
                                                      "tag": "path",
                                                    },
                                                  ],
                                                  "tag": "svg",
                                                },
                                                "name": "right",
                                                "theme": "outlined",
                                              }
                                            }
                                          >
                                            <span
                                              aria-label="right"
                                              className="anticon anticon-right"
                                              role="img"
                                            >
                                              <IconReact
                                                icon={
                                                  Object {
                                                    "icon": Object {
                                                      "attrs": Object {
                                                        "focusable": "false",
                                                        "viewBox": "64 64 896 896",
                                                      },
                                                      "children": Array [
                                                        Object {
                                                          "attrs": Object {
                                                            "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                                          },
                                                          "tag": "path",
                                                        },
                                                      ],
                                                      "tag": "svg",
                                                    },
                                                    "name": "right",
                                                    "theme": "outlined",
                                                  }
                                                }
                                              >
                                                <svg
                                                  aria-hidden="true"
                                                  data-icon="right"
                                                  fill="currentColor"
                                                  focusable="false"
                                                  height="1em"
                                                  key="svg-right"
                                                  viewBox="64 64 896 896"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                                                    key="svg-right-svg-0"
                                                  />
                                                </svg>
                                              </IconReact>
                                            </span>
                                          </AntdIcon>
                                        </ForwardRef(RightOutlined)>
                                      </button>
                                    </li>
                                  </ul>
                                </Pagination>
                              </LocaleReceiver>
                            </Pagination>
                          </Paginator>
                        </div>
                      </main>
                    </Main>
                  </Posts>
                </Connect(Posts)>
              </Route>
            </Switch>
          </Router>
        </HashRouter>
        <Footer>
          <footer
            className="footer"
          >
            <div
              className="container"
            >
              <div
                className="copyright"
              >
                Delta World © 1970-2077
              </div>
              <LanguageSelect>
                <select
                  className="language"
                  id="language"
                  name="language"
                  onChange={[Function]}
                  value="ru"
                >
                  <option
                    value="ru"
                  >
                    RU
                  </option>
                  <option
                    value="en"
                  >
                    EN
                  </option>
                </select>
              </LanguageSelect>
              <Theme>
                <div
                  className="theme"
                >
                  <span
                    className="text"
                  >
                    Тёмная тема
                  </span>
                  <Switch
                    checked={false}
                    onChange={[Function]}
                  >
                    <Wave
                      insertExtraNode={true}
                    >
                      <Switch
                        checked={false}
                        className=""
                        loadingIcon={
                          <div
                            className="ant-switch-handle"
                          />
                        }
                        onChange={[Function]}
                        prefixCls="ant-switch"
                      >
                        <button
                          aria-checked={false}
                          className="ant-switch"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="switch"
                          type="button"
                        >
                          <div
                            className="ant-switch-handle"
                          />
                          <span
                            className="ant-switch-inner"
                          />
                        </button>
                      </Switch>
                    </Wave>
                  </Switch>
                </div>
              </Theme>
            </div>
          </footer>
        </Footer>
      </div>
    </ThemeContextProvider>
  </App>
</Provider>
`;
